modules/legacy.js

```diff
@@ -24,112 +72,51 @@ const proxyLegacyPage = ({ req, res, domModifications, followRedirect = true })
-            let titleText = dom.window.document.title;
-            dom.window.document.title = '[PROXIED] ' + titleText;
-        }
+        /**
+         * Always redirect to canonical link
+         * Sitecore serves the content of vanity URLs rather
+         * than redirecting, so to avoid duplicate content and make it
+         * easier to replace old URLs we attempt to look up the canonical URL
+         * from the page and redirect to that.
+         */
+        const canonicalUrl = getCanonicalUrl(dom);
+        if (canonicalUrl && canonicalUrl !== req.path) {
+            res.redirect(301, canonicalUrl);
+        } else {
+            if (appData.isDev) {
+                dom.window.document.title = '[PROXIED] ' + dom.window.document.title;
+            }
 
-        const regionNav = dom.window.document.getElementById('regionNav');
-        const pageIsWelsh = isWelsh(req.path);
-        const hasWelshLink = regionNav.querySelectorAll('[hreflang="cy"]').length > 0;
-        const hasEnglishLink = regionNav.querySelectorAll('[hreflang="en"]').length > 0;
-
-        const appendLanguageLink = (localeToAppend, nav) => {
-            const linkPath = localeToAppend === 'cy' ? makeWelsh(req.path) : removeWelsh(req.path);
-            const linkText = localeToAppend === 'cy' ? 'Cymraeg' : 'English';
-            const listItem = dom.window.document.createElement('li');
-            listItem.setAttribute('id', 'ctl12_langLi');
-            listItem.setAttribute('class', 'last');
-            listItem.innerHTML = `
-                <a href="${linkPath}"
-                   id="ctl12_welshLanguage"
-                   lang="${localeToAppend}"
-                   hreflang="${localeToAppend}"
-                   data-blf-alpha="true">
-                   ${linkText}
-               </a>`;
-            nav.appendChild(listItem);
-        };
-
-        // some pages aren't welsh but *do* have the link
-        // other pages aren't welsh but don't (due to cookies etc)
-        if (!pageIsWelsh && !hasWelshLink) {
-            appendLanguageLink('cy', regionNav);
-        } else if (pageIsWelsh && !hasEnglishLink) {
-            appendLanguageLink('en', regionNav);
-        }
+            appendLanguageLink(req, res, dom);
 
-        // rewrite main ASP.net form to point to this page
-        // (currently it's rewritten above to the external one)
-        const form = dom.window.document.getElementById('form1');
-        if (form) {
-            form.setAttribute('action', req.path);
-        }
+            // rewrite main ASP.net form to point to this page
+            // (currently it's rewritten above to the external one)
+            const form = dom.window.document.getElementById('form1');
+            if (form) {
+                form.setAttribute('action', req.path);
+            }
 
-        /**
-         * Are we in an A/B test and do we have a Google Experiments ID
-         */
-        const experimentId = get(res.locals, 'ab.id');
```

Comment:
I removed this whole analytics injection block on the basis that we don't have any A/B tests against legacy pages, and we're unlikely to again. AFAIK this was mostly for the homepage rollout right?

Reply:
Yeah you're right, we shouldn't need to do it again either.